persistence/types/gov.go

```diff
@@ -42,6 +43,36 @@ const (
 		)`
 )
 
+type ParamOrFlag struct {
```

Comment:
The solution you have works, but I don't think it's great that the structure and serialization of what determines the state hash is embedded this deep in the code and is golang specific.

Fwiw, I don't think our current code structure is good at explaining this, but the idea is that we're using protobufs (a universal serialization codec) to maintain what the state hash is. This will enable other implementations (in other languages) to implement versions of the Pocket protocol as well.

1. See this diagram in the persistence spec for more details: https://github.com/pokt-network/pocket-network-protocol/tree/main/persistence#521-state-hash-computation

2. For now, can you create a new protobuf under `shared/core/types/proto` called `params.go`, and have a `message Param {...}` and `message Flag {...}`, so the "Hash" implementation you have below will be a proto marshalling of the object.

Reply:
I think I have implemented this correctly but I've not used Protobufs before so my implementation could definitely be improved upon - I am using `[]byte(param.String())` and `proto.Marshal(param)` as the key and values in the tree now same for flag